sherpa/models/model.py

```diff
@@ -536,6 +536,8 @@ class provides case-insensitive access to the parameter names
     def __init__(self,
                  name: str,
                  pars: Sequence[Parameter] = ()) -> None:
+        if not isinstance(name, str):
+            raise TypeError("The parameter 'name' for a model must be a string but it is a " + str(type(name)) + ".")
```

Comment:
Question: It is OK to raise a Python error here since it's not in the UI or should I pick something from the `sherpa.utils.err` module instead?

Reply:
If you think it's OK, I'll leave it. I like the way I can format the message to the users right here.